### ISSUE TITLE
`JoinError` when commiting a pessimistic transaction and quickly exiting

### ISSUE THREAD

**Issue by johnbchron** [tikv/client-rust#463]
I frequently run into `JoinError` from [these lines](https://github.com/tikv/client-rust/blob/5a8a3c4cccffbe87bfeb866a982f5b2eaee04736/src/transaction/transaction.rs#L1299-L1303) when I commit a pessimistic transaction and quickly end the program.

I don't mean terminate the program, but something like this, where `db.migrate()` creates and commits pessimistic transactions:
```rust
#[tokio::main]
async fn main() -> Result<()> {
  tracing_subscriber::fmt::init();

  let db = db::TikvDb::new().await?;
  db.migrate().await?;

  Ok(())
}
```

It seems like not a good idea to have no way for the consumer to wait for these tasks to finish. My workaround for the moment is to simply have tokio idle the runtime for 1 second while I wait for those tasks to finish in the background.

Ideally the `txn.commit().await` would join the tasks it spawns.

**Comment by pingyu** [comment]
`txn.comit().await` does not join the tasks of committing secondaries keys by design, to reduce the latency of clients.

Can you provide the backtrace about where the `JoinError` is raised from ? 

(p.s. Change the `?` to `unwrap()` may help).

**Comment by johnbchron** [comment]
I don't get it from an error (even when I unwrap); instead I just get this log line:

```
2024-09-16T16:49:31.835013Z  WARN tikv_client::transaction::transaction: Failed to commit secondary keys: Failed to join tokio tasks
```

Which seems to come from [here](https://github.com/tikv/client-rust/blob/f6774b46a863f012c3a5bb9b1dc5540747082cf2/src/transaction/transaction.rs#L1301). I assumed that it was generating (and catching) a join error.

As far as I can tell, all of the writes I performed went through, but it's still unsettling. Maybe we could add some method in addition to `commit()` that makes sure that everything happens, at the cost of speed?

**Comment by pingyu** [comment]
> I don't get it from an error (even when I unwrap); instead I just get this log line:
> 
> ```
> 2024-09-16T16:49:31.835013Z  WARN tikv_client::transaction::transaction: Failed to commit secondary keys: Failed to join tokio tasks
> ```
> 
> Which seems to come from [here](https://github.com/tikv/client-rust/blob/f6774b46a863f012c3a5bb9b1dc5540747082cf2/src/transaction/transaction.rs#L1301). I assumed that it was generating (and catching) a join error.
> 
> As far as I can tell, all of the writes I performed went through, but it's still unsettling. Maybe we could add some method in addition to `commit()` that makes sure that everything happens, at the cost of speed?

We do not try to join the task here, so it should not meet the join error.

I think the error was raise from [`single_plan_handler`](https://github.com/tikv/client-rust/blob/f6774b46a863f012c3a5bb9b1dc5540747082cf2/src/request/plan.rs#L133) where try to join tasks from all shards.

And this warning can be safely ignored. It does not break correctness, the uncommitted records of secondary keys will be resolved on the next read or write of the keys.

**Comment by johnbchron** [comment]
I see. Thanks for the explanation.
